pkg/filter/dt/filter_mysql_query.go

```diff
@@ -105,27 +112,28 @@ func (f *_mysqlFilter) processAfterQueryDelete(ctx context.Context, conn *driver
 	log.Debugf("delete, lockKey: %s", lockKeys)
 	undoLog := exec.BuildUndoItem(false, constant.SQLType_DELETE, schemaName, executor.GetTableName(), lockKeys, biValue, nil)
 
-	branchID, err := f.registerBranchTransaction(ctx, xid, conn.DataSourceName(), lockKeys)
+	branchID, err := f.registerBranchTransaction(newCtx, xid, conn.DataSourceName(), lockKeys)
 	if err != nil {
 		return err
 	}
 	log.Debugf("delete, branch id: %d", branchID)
 	return dt.GetUndoLogManager().InsertUndoLogWithNormal(conn, xid, branchID, undoLog)
 }
 
-func (f *_mysqlFilter) processAfterQueryInsert(ctx context.Context, conn *driver.BackendConnection,
-	result proto.Result, insertStmt *ast.InsertStmt) error {
+func (f *_mysqlFilter) processAfterQueryInsert(ctx context.Context, conn *driver.BackendConnection, result proto.Result, insertStmt *ast.InsertStmt) error {
 	has, xid := misc.HasXIDHint(insertStmt.TableHints)
+	newCtx, span := tracing.GetTraceSpan(ctx, "mysql_process_after_query_insert")
```

Comment:
I recommend not adding TraceContext for this location. instead, add inside of func `executor.AfterImage`:
```
newCtx, span := tracing.GetTraceSpan(ctx, "fetch_after_image")
```